build/Common.props

```diff
@@ -1,20 +1,51 @@
 <Project>
   <PropertyGroup>
     <LangVersion>8.0</LangVersion>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)debug.snk</AssemblyOriginatorKeyFile>
+    <DefineConstants>$(DefineConstants);SIGNED</DefineConstants>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
@reyang @cijothomas @eddynaka How do you guys feel about adding `Condition="'$(Configuration)'=='Release'"` to `TreatWarningsAsErrors`? The idea is that way it will fail CI if a warning is introduced but while doing dev/test loop you can get your code working before having to make everything happy.

Reply:
I think that's good. We could do some script to execute. Something like build+test in release mode. so u don't need to it manually.

Reply:
Instead of making it conditional based on `Release` , can we instead get a single setting which can turn this off? So if people are distracted by this requirement while development, they can disable it.

Running it only for 'Release' means - most of the time, people don't realize something is broken until CI catches it, instead of catching it early and avoid unwanted round trips..

Reply:
I'm a big fan of running the analyzers in Release only mode.
I agree with Cijo that it can be troublesome for PRs, but I haven't found anything better. 
As long as the builds are public, authors can review why their PRs failed.

Here are all the settings we use in Application Insights:
```
<!-- Microsoft recommends that we disable CodeAnalysis in favor of FxCop Analyzers. aka.ms/fxcopanalyzers -->
<RunCodeAnalysis>false</RunCodeAnalysis>
<CodeAnalysisTreatWarningsAsErrors>false</CodeAnalysisTreatWarningsAsErrors>

<StyleCopEnabled>true</StyleCopEnabled>
<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
<StyleCopTreatErrorsAsWarnings>true</StyleCopTreatErrorsAsWarnings>


<!--This is to disable code analysis while devs are working. Default is true.-->
<RunAnalyzers Condition=" '$(Configuration)' == 'Debug' ">false</RunAnalyzers>
```

https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/.props/_AnalyzerSettings.props
https://github.com/microsoft/ApplicationInsights-dotnet/blob/911a5cc7b99121b47590d6238ecb0abbbef2793d/Directory.Build.props#L39-L40

Reply:
Like the idea 👍.
I think CI needs to run for both Debug and Release.
Regarding CI capturing more issues than the local devbox, I think that is totally fine (as long as there is a way for people to easily see the error and have a local repro).